src/main/java/com/mael/ttt/ui/MenuOption.java

```diff
@@ -22,14 +22,21 @@ public String getMenuOptionText() {
         return menuOptionText;
     }
 
-
-    public static MenuOption idToOption(String menuOptionId) {
+    public static MenuOption idToOption(String id) {
         for (MenuOption menuOption : values()) {
-            if (menuOptionId.equals(menuOption.getMenuOptionId())) {
+            if (id != null && id.equals(menuOption.getMenuOptionId())) {
```

Comment:
Flipping the check enables you to get rid of the `null` check, i.e. something like `if (menuOption.getMenuOptionId().equals(id)) {`.
_If_ `id` is null this `equals` check will return `false` automatically.

As a side note I think `getMenuOptionId()` and `getMenuOptionText()` contains duplication in the name. These are methods of `MenuOption` already, so I'd probably just name them `getId()` and `getText()`.